app/post.go

```diff
@@ -1353,6 +1490,10 @@ func (a *App) SearchPostsForUser(c *request.Context, terms string, userID string
 		}
 	}
 
+	if appErr := a.filterInaccessiblePosts(postSearchResults.PostList); appErr != nil {
+		return nil, model.NewAppError("SearchPostsForUser", "app.post.filter_inaccessible_posts.app_error", nil, appErr.Error(), http.StatusInternalServerError)
```

Comment:
Just return the error directly. No need to unwrap-wrap again. It's already an AppError.